processes-services/src/main/java/org/exoplatform/processes/service/ProcessesAttachmentServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+
+  private final NodeFinder             nodeFinder;
+
+  private static final String    PROCESSES_SPACE_GROUP_ID = "/spaces/processes_space";
+
+  public ProcessesAttachmentServiceImpl(AttachmentService attachmentService,
+                                        RepositoryService repositoryService,
+                                        SessionProviderService sessionProviderService,
+                                        ManageDriveService manageDriveService,
+                                        NodeHierarchyCreator nodeHierarchyCreator,
+                                        NodeFinder nodeFinder) {
+    this.attachmentService = attachmentService;
+    this.repositoryService = repositoryService;
+    this.sessionProviderService = sessionProviderService;
+    this.manageDriveService = manageDriveService;
+    this.nodeHierarchyCreator = nodeHierarchyCreator;
+    this.nodeFinder = nodeFinder;
+  }
+
+  @Override
+  public void linkAttachmentsToEntity(Attachment[] attachments, Long userId, Long entityId, String entityType) {
+    if (attachments != null && attachments.length > 0) {
+      Arrays.stream(attachments).map(Attachment::getId).forEach(attachmentId -> {
+        try {
+          attachmentService.linkAttachmentToEntity(userId, entityId, entityType, attachmentId);
+        } catch (Exception e) {
+          LOG.error("Error while attaching files to entity", e);
+        }
+      });
+      moveOrCopyAttachmentsJcrNodes(Arrays.asList(attachments), entityId, entityType, false);
+    }
+  }
+
+  private void linkFromEntityToEntity(Long userId,
+                                      List<Attachment> attachments,
+                                      Long sourceEntityId,
+                                      String sourceEntityType,
+                                      Long destEntityId,
+                                      String destEntityType,
+                                      boolean deleteSource) {
+    attachments.stream().map(Attachment::getId).forEach(attachmentId -> {
+      try {
+        attachmentService.linkAttachmentToEntity(userId, destEntityId, destEntityType, attachmentId);
+      } catch (Exception e) {
+        LOG.error("Error while attaching files to entity", e);
+      }
+    });
+    if (attachments.size() != 0 && deleteSource) {
+      try {
+        attachmentService.deleteAllEntityAttachments(userId, sourceEntityId, sourceEntityType);
+      } catch (Exception e) {
+        LOG.error("Error while removing attachments to entity source", e);
+      }
+    }
+
+  }
+  
+  private void moveOrCopyAttachmentsJcrNodes(List<Attachment> attachments, Long destEntityId, String destEntityType, boolean copy) {
+    Map<String, String[]> permissions = new HashMap<>();
+    Map<Integer, Attachment>  updatedAttachments = new HashMap<>();
+    permissions.put("*:/platform/administrators", PermissionType.ALL);
+    permissions.put("*:/platform/processes", PermissionType.ALL);
```

Comment:
Should not set hardcoded permissions, we may have cases  that process are created in other space projects or maybe personal projects